### PR TITLE
Add index on log.dttm

### PR DESCRIPTION
**Pull request by pingzh**
log table tends to be very large. Having an index around `dttm` can greatly improve the query performance when trying to filter log records in a period of time

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/migrations/versions/6abdffdd4815_add_dttm_index_on_log_table.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# revision identifiers, used by Alembic.
+revision = "6abdffdd4815"
+down_revision = "290244fb8b83"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """Apply add dttm index on log table"""
+    op.create_index("idx_log_dttm", "log", ["dttm"], unique=False)
```
Comment (kaxil) [inline airflow/migrations/versions/6abdffdd4815_add_dttm_index_on_log_table.py]: You should add it on the following line too:

https://github.com/apache/airflow/blob/911b708ffddd4e7cb6aaeac84048291891eb0f1f/airflow/models/log.py#L42-L45

File: airflow/migrations/versions/6abdffdd4815_add_dttm_index_on_log_table.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    pass
```
Comment (kaxil) [inline airflow/migrations/versions/6abdffdd4815_add_dttm_index_on_log_table.py]: not needed

```suggestion
```

File: airflow/migrations/versions/6abdffdd4815_add_dttm_index_on_log_table.py (hunk 1 above)
Comment (pingzh) [reply airflow/migrations/versions/6abdffdd4815_add_dttm_index_on_log_table.py]: ag, good catch, thanks

File: airflow/migrations/versions/0123_2_6_0_add_dttm_index_on_log_table.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from __future__ import annotations
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "6abdffdd4815"
+down_revision = "290244fb8b83"
+branch_labels = None
+depends_on = None
+airflow_version = "2.6.0"
```
Comment (pingzh) [inline airflow/migrations/versions/0123_2_6_0_add_dttm_index_on_log_table.py:35]: @kaxil do you think we can release it for version `2.5.1`?

Comment (kaxil) [reply airflow/migrations/versions/0123_2_6_0_add_dttm_index_on_log_table.py:35]: Too late for 2.5.1 I am afraid

**Review by kaxil — APPROVED** [review]

File: airflow/migrations/versions/0123_2_6_0_add_dttm_index_on_log_table.py (hunk 3 above)
Comment (pingzh) [reply airflow/migrations/versions/0123_2_6_0_add_dttm_index_on_log_table.py:35]: 👍

**Pull request merged**